d3a_interface/matching_algorithms/pay_as_clear_matching_algorithm.py

```diff
@@ -0,0 +1,212 @@
+"""
+Copyright 2018 Grid Singularity
+This file is part of D3A.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+
+import math
+from logging import getLogger
+from collections import OrderedDict
+from typing import List, Dict, Tuple
+
+from d3a_interface.constants_limits import ConstSettings
+from d3a_interface.dataclasses import MarketClearingState, Clearing, BidOfferMatch
+from d3a_interface.matching_algorithms import BaseMatchingAlgorithm
+from d3a_interface.utils import sort_list_of_dicts_by_attribute, add_or_create_key
+
+log = getLogger(__name__)
+
+MATCH_FLOATING_POINT_TOLERANCE = 1e-8
```

Comment:
Why not using [FLOATING_POINT_TOLERANCE](https://github.com/gridsingularity/d3a-interface/blob/master/d3a_interface/constants_limits.py#L246) and possibly update the precision level ?

Reply:
I did not touch the code here, however these two have different value AFAIS

Reply:
Leave it as it is for now, but do look in your next enhancement ticket if these two closely related constants could be merged